src/shared/util.js

```diff
@@ -567,15 +567,19 @@ class AbortException extends BaseException {
 }
 
 const NullCharactersRegExp = /\x00/g;
+const InvisibleCharactersRegExp = /[\x00-\x1F]/g;
 
 /**
  * @param {string} str
  */
-function removeNullCharacters(str) {
+function removeNullCharacters(str, replaceInvisible = false) {
   if (typeof str !== "string") {
     warn("The argument for removeNullCharacters must be a string.");
     return str;
   }
+  if (replaceInvisible) {
+    return str.replace(InvisibleCharactersRegExp, " ");
```

Comment:
My suggestion was to *purposely* leave the `\x00`-case intact, even in the new mode, since there could very well be regressions by changing that.
Not to mentioned that having the `\x00`-char treated differently depending on the value of the new parameter feels surprising.

In the PDF document that originally prompted us to add the `removeNullCharacters`-function, I believe that this will lead to a bunch of trailing spaces, which could also potentially affect formatting of the outline-titles.

Reply:
Don't you think we should trim the outline title when rendering it ?